src/app/modules/payments/stripe.controller.ts

```diff
@@ -91,84 +68,157 @@ export const _handleStripeEventUpdates: ControllerHandler<
+
   switch (event.type) {
-    case 'charge.captured':
-    case 'charge.expired':
+    // We catch all payment_intent, charge and payout events, except the
+    // following ignored event types (as associated features are not supported):
+    // - charge.captured, charge.expired (only for capture flow)
+    // - charge.updated (descriptions or metadata are updated)
+    // - payment_intent.amount_capturable_updated (only for capture flow)
+    // - payment_intent.partially_funded (only occurs when payment intents are
+    //   completed in part by customer stripe account balance)
+    case 'payment_intent.canceled':
+    case 'payment_intent.created':
+    case 'payment_intent.payment_failed':
+    case 'payment_intent.processing':
+    case 'payment_intent.requires_action':
+    case 'payment_intent.succeeded':
     case 'charge.failed':
     case 'charge.pending':
     case 'charge.refunded':
-    case 'charge.succeeded':
-    case 'charge.updated':
-      await findPaymentAndUpdate(
+    case 'charge.succeeded': {
+      result = await getMetadataPaymentId(
         event.data.object.metadata,
-        {
-          chargeIdLatest: event.data.object.id,
-          status: stripeChargeStatusToPaymentStatus(event.data.object.status),
-        },
-        event,
+      ).asyncAndThen((paymentId) =>
+        StripeService.processStripeEvent(paymentId, event),
       )
       break
+    }
     case 'charge.dispute.closed':
     case 'charge.dispute.created':
     case 'charge.dispute.funds_reinstated':
     case 'charge.dispute.funds_withdrawn':
-    case 'charge.dispute.updated':
-      await findPaymentAndUpdate(event.data.object.metadata, {}, event)
-      break
-    case 'charge.refund.updated':
-      // We should actually handle this case, need to implement based on get by charge Id though, not doing for hackathon.
+    case 'charge.dispute.updated': {
+      const chargeIdLatest =
+        typeof event.data.object.charge === 'string'
+          ? event.data.object.charge
+          : event.data.object.charge.id
+
+      const payment = await PaymentModel.findOne({ chargeIdLatest })
+      if (!payment) {
+        logger.warn({
+          message: 'Received dispute event with unknown latest charge id',
+          meta: { ...logMeta, chargeIdLatest },
+        })
+        return res.sendStatus(StatusCodes.BAD_REQUEST)
+      }
+
+      result = await StripeService.processStripeEvent(payment.id, event)
       break
-    case 'payment_intent.amount_capturable_updated':
-    case 'payment_intent.canceled':
-    case 'payment_intent.created':
-    case 'payment_intent.partially_funded':
-    case 'payment_intent.payment_failed':
-    case 'payment_intent.processing':
-    case 'payment_intent.requires_action':
-    case 'payment_intent.succeeded':
-      await findPaymentAndUpdate(event.data.object.metadata, {}, event)
+    }
+    case 'charge.refund.updated': {
+      const chargeIdLatest =
+        typeof event.data.object.charge === 'string'
```

Comment:
This extraction of chargeId from `object.charge` is scattered in quite a few places, I think it makes sense to make it into a utils func